### PR TITLE
chart: install components with host mode

### PR DESCRIPTION
**Pull request by calvin0327**
Signed-off-by: calvin0327 <wen.chen@daocloud.io>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
install together karmada and its component together use host installMode.
this is test result:
image

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/2290

**Special notes for your reviewer**:
please correct me if not.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by calvin0327** [comment]
@Poor12 PTAL

**Comment by Poor12** [comment]
Hi @calvin0327, you can pull and rebase this pr to retrigger the CI.

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,6 @@ _output/
 
 # OSX trash
 .DS_Store
+
+# sub chart tgz
+charts/karmada/charts
```
Comment (RainbowMango) [inline .gitignore:26]: Should we remove the path `https://github.com/karmada-io/karmada/tree/master/charts/karmada/charts` ? it's already in the repo.

Comment (calvin0327) [reply .gitignore:26]: This is the package we depency on, which is a tgz file. When we pack chart, we can exec `helm package ./charts/karmada --dependency-update` to automatically download it for us.
I think we should not upload tgz file to our repo.

Comment (calvin0327) [reply .gitignore:26]: we can delete the file `https://github.com/karmada-io/karmada/tree/master/charts/karmada/charts` together.

Comment (Poor12) [reply .gitignore:26]: Yeah, it has been already in our repo. Can you delete it together in this pr?

Comment (calvin0327) [reply .gitignore:26]: done

Comment (Poor12) [reply .gitignore:26]: Hi @calvin0327, could you help check why chart-lint failed?

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**